packages/perseus/src/util/snowman-utils.test.ts

```diff
@@ -0,0 +1,93 @@
+import {
+    addWidget,
+    getAllWidgetIds,
+    WidgetType,
+    getQuestionWidgetIds,
+    getAllWidgetTypes,
+} from "./snowman-utils";
+
+describe("getWidgetTypes", () => {
+    it("returns all widget types present in a string with no duplicates", () => {
+        const content =
+            "> [[☃ passage 3]]\n\n=====\n\n###Questions for Passage 1\n\n[[☃ group 1]]  \n[[☃ group 12]]  \n\n$$\n";
+        const actual = getAllWidgetTypes(content);
+        const expected = ["passage", "group"];
+        expect(actual).toEqual(expected);
+    });
+});
+
+describe("getWidgetNamesFromContent", () => {
+    it("returns all widget names present in a string", () => {
+        const content =
+            "> [[☃ passage 3]]\n\n=====\n\n###Questions for Passage 1\n\n[[☃ group 1]]  \n[[☃ group 12]]  \n\n$$\n";
+        const actual = getAllWidgetIds(content);
+        const expected = ["passage 3", "group 1", "group 12"];
+        expect(actual).toEqual(expected);
+    });
+});
+
+describe("getAllWidgetIds", () => {
+    it("returns all widget names present in a string", () => {
```

Comment:
nit: should this be widget ids for consistency with the test names? (non blocking)

Reply:
Ooo, thank you so much! :D